float32 velocity
float32 acceleration
